SPECS/python-pycryptodome/python-pycryptodome.spec

```
@@ -0,0 +1,419 @@
+# published by the Open Source Initiative.
+
+# Please submit bugfixes or comments via https://bugs.opensuse.org/
+#
+
+%{?!python_module:%define python_module() python-%{**} python3-%{**}}
+%define oldpython python
+Name:           python-pycryptodome
+Version:        3.14.1
+Release:        3
+Summary:        Cryptographic library for Python
+License:        BSD-2-Clause
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          Development/Languages/Python
+URL:            https://www.pycryptodome.org
+Source:         https://github.com/Legrandin/pycryptodome/archive/v%{version}.tar.gz#/%{name}-%{version}.tar.gz
+BuildRequires:  python3-devel
+BuildRequires:  python3-setuptools
+BuildRequires:  fdupes
+BuildRequires:  python-rpm-macros
+%if !0%{?_no_weakdeps}
+# PyCryptodome uses gmp via cffi as runtime optimization
+# would be better, if libgmp* would provide gmp
+Suggests:       libgmp10
+Suggests:       python-cffi
+%endif
+Provides:       python-pycrypto = %{version}
+Obsoletes:      python-pycrypto < %{version}
+
+%description
+PyCryptodome is a self-contained Python package of low-level
+cryptographic primitives.
+
+PyCryptodome is a fork of PyCrypto, residing in the `Crypto`
+namespace for better drop-in compatibility, while it brings several
+enhancements with respect to the last official version of PyCrypto
+(2.6.1), for instance:
+
+* Authenticated encryption modes (GCM, CCM, EAX, SIV, OCB)
+* Accelerated AES on Intel platforms via AES-NI
+* First class support for PyPy
+* Elliptic curves cryptography (NIST P-256 curve only)
+* Better and more compact API (`nonce` and `iv` attributes for
+  ciphers, automatic generation of random nonces and IVs, simplified
+  CTR cipher mode, and more)
+* SHA-3 (including SHAKE XOFs), SHA-512/t and BLAKE2 hash algorithms
+* Salsa20 and ChaCha20 stream ciphers
+* Poly1305 MAC
+* ChaCha20-Poly1305 authenticated cipher
+* scrypt and HKDF
+* Deterministic (EC)DSA
+* Password-protected PKCS#8 key containers
+* Shamir's Secret Sharing scheme
+* Random numbers get sourced directly from the OS (and not from a
+  CSPRNG in userspace)
+* Simplified install process, including better support for Windows
+* Cleaner RSA and DSA key generation (largely based on FIPS 186-4)
+* Major clean ups and simplification of the code base
+
+PyCryptodome is not a wrapper to a separate C library like *OpenSSL*.
+To the largest possible extent, algorithms are implemented in pure
+Python. Only the pieces that are extremely critical to performance
+(e.g. block ciphers) are implemented as C extensions.
+
+%prep
+%autosetup -n pycryptodome-%{version}
+
+%build
+export LC_ALL=en_US.UTF-8
+export CFLAGS="%{optflags}"
+%py3_build
+
+%install
+export LC_ALL=en_US.UTF-8
+%py3_install
+%fdupes %{buildroot}%{$python3_sitearch}
+
+%check
+export LC_ALL=en_US.UTF-8
+%{pushd %{buildroot}%{python3_sitearch}
+PYTHONPATH=%{buildroot}%{python3_sitearch} $python -m Crypto.SelfTest
+popd}
```

Not a huge fan of this syntax for several reasons- I'm wary of wrapping the whole block in a macro since you don't see that often, I'm not sure `$python` is defined in the shell here, and the output of is masked by the fact that it's not the last command ran in this section.

I'd suggest the following, but please check that it still works.

```suggestion
cd %{buildroot}%{python3_sitearch}
PYTHONPATH=. %python3 -m Crypto.SelfTest
```